compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -93,6 +94,53 @@ pub(super) fn search_for_section<'a>(
         .map_err(|e| format!("failed to read {} section in '{}': {}", section, path.display(), e))
 }
 
+fn add_gnu_property_note(
+    file: &mut write::Object<'static>,
+    architecture: Architecture,
+    binary_format: BinaryFormat,
+    endianness: Endianness,
+) {
+    // check bti protection
+    if binary_format != BinaryFormat::Elf
+        || !matches!(architecture, Architecture::X86_64 | Architecture::Aarch64)
```

Comment:
I think it's also relevant to `Architecture::I386` and `X86_64_X32`, and maybe `Aarch64_Ilp32` too.

Reply:
`I386` will not be implemented... in the kernel? I've seen that `gcc -m32 -fcf-protection` does write this already.